File: Classes/Extensions/UIKit/UIApplication+Helper.m
//
//  UIApplication+Helper.m
//  iSub
//
//  Created by Benjamin Baron on 11/9/20.
//  Copyright © 2020 Ben Baron. All rights reserved.
//

#import "UIApplication+Helper.h"

@implementation UIApplication (Helper)

+ (UIInterfaceOrientation)orientation {
    return [[[[[UIApplication sharedApplication] windows] firstObject] windowScene] interfaceOrientation];
}

+ (UIWindow *)keyWindow {
    for (UIWindow *window in [[UIApplication sharedApplication] windows]) {
        if (window.isKeyWindow) return window;
    }
    return nil;
}

+ (CGFloat)statusBarHeight {
    return self.keyWindow.windowScene.statusBarManager.statusBarFrame.size.height;
}

@end

File: Classes/Extensions/UIKit/UIApplication+Helper.swift
//
//  UIApplication+Helper.swift
//  iSub
//
//  Created by Benjamin Baron on 1/19/21.
//  Copyright © 2021 Ben Baron. All rights reserved.
//

import UIKit

extension UIApplication {
    static var orientation: UIInterfaceOrientation {
        UIApplication.shared.windows.first?.windowScene?.interfaceOrientation ?? .portrait
    }
    
    static var keyWindow: UIWindow? {
        UIApplication.shared.windows.first { $0.isKeyWindow }
    }
    
    static var statusBarHeight: CGFloat {
        keyWindow?.windowScene?.statusBarManager?.statusBarFrame.size.height ?? 0
    }
}
